### DISCUSSION TITLE
Print to PDF on iphone

### DISCUSSION THREAD

**Discussion by martingeh** [opengisch/QField#3854] [Q&A]
Hi,  I've managed to get my project from desktop to my iphone and have made sure to include a print layout or two in the project file.  On the iphone, I don't get the option to print to pdf, just print to image.
I select that, then select the layout and it shows as an image, but without a basemap and layers, even though I can see the basemap and layers etc. in Qfield itself.  
I'm using iphone 13 Pro Max, iOS vers16.3 and using Qfield 2.6.2.
[Print to PDF.pdf](https://github.com/opengisch/QField/files/10529491/Print.to.PDF.pdf)

**Comment by nirvn** [comment]
@martingeh , gut feeling: your layout map item CRS doesn't match that of the project. You can check this here in your layout:

![image](https://user-images.githubusercontent.com/1728657/215319103-6fbf39ba-5725-4eac-bc96-586eb73e1e0a.png)

Also, make sure your reference map is set to your main map item that you want driven by QField:

![image](https://user-images.githubusercontent.com/1728657/215319144-30fbe983-81e5-49fc-8c73-7cfed66da351.png)

**Reply by martingeh** [reply]
Right - will have a look at that in a few hours. Can't say I checked that before, so you're likely to be right there - thanks. Will reply again after checking.

**Reply by martingeh** [reply]
Thankyou very much @nirvn!  Such a simple thing that I didn't check. That works as expected. 
Would you know why it prints to an image instead of PDF?  I have a workaround to get it into a PDF, but of course it would be more streamlined if it printed straight to PDF. 

Thanks again Mathieu.

**Reply by nirvn** [reply]
@martingeh , writing PDF requires printing support within upstream Qt library, and that's not available on iOS. So on that platform, we instead produce images.

There's an open Qt issue on this: https://bugreports.qt.io/plugins/servlet/mobile#issue/QTBUG-34555 -- by the looks of it we're not going to see it supported anytime soon.

**Reply by martingeh** [reply]
Okay, not a big issue really (for me). Thanks for the reply Mathieu.
